test/utils/image/manifest.go

```diff
@@ -265,3 +272,38 @@ func (i *Config) GetE2EImage() string {
 func GetPauseImageName() string {
 	return GetE2EImage(Pause)
 }
+
+// ReplaceRegistryInImageURL replaces the registry in the image URL with a custom one
+func ReplaceRegistryInImageURL(imageURL string) (string, error) {
+	parts := strings.Split(imageURL, "/")
+	countParts := len(parts)
+	registryAndUser := strings.Join(parts[:countParts-1], "/")
+
+	switch registryAndUser {
+	case "gcr.io/kubernetes-e2e-test-images":
+		registryAndUser = e2eRegistry
+	case "k8s.gcr.io":
+		registryAndUser = gcRegistry
+	case "gcr.io/k8s-authenticated-test":
+		registryAndUser = PrivateRegistry
+	case "gcr.io/google-samples":
+		registryAndUser = sampleRegistry
+	case "docker.io/library":
+		registryAndUser = dockerLibraryRegistry
+	case "gcr.io/gke-release":
+		registryAndUser = gcrReleaseRegistry
+	case "quay.io/k8scsi":
+		registryAndUser = quayK8sCSI
+	default:
+		if countParts == 1 {
```

Comment:
Do we already check if registryAndUser is a valid registry/user string? The `countParts == 1` feels a bit too optimistic to me.

Reply:
AFAIK currently there is no check if `registry/user` is actually a valid registry. I think this is also the default behaviour of the docker daemon (without judging if this is a good idea)